### PR TITLE
Adds convenience methods for using the Syft CLI to create SBoM information

### PR DESCRIPTION
**Pull request by dmikusa-pivotal**
## Summary

- Supports running syft against a directory & generating one or more output formats
- Automatically converts syft generated CycloneDX XML to JSON, which is what buildpacks require
- This is working around a couple features not present in syft at the moment, such as support for multiple output formats or conversion, and being able to output directly to CycloneDX JSON. When those features are added, we can trim this code back accordingly. The function signatures should not need to change for that.

## Checklist

* [ ] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [ ] I have added an integration test, if necessary.
* [ ] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [ ] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Comment by dmikusa-pivotal** [comment]
Tests are failing because this depends on https://github.com/buildpacks/libcnb/pull/95.

**Comment by samj1912** [comment]
Related - https://github.com/anchore/syft/issues/631

File: sherpa/sbom.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+	if backup {
+		if err := b.backupXMLFile(inputPath); err != nil {
+			return fmt.Errorf("unable to backup file\n%w", err)
+		}
+	}
+
+	bom, err := b.readXMLBOM(inputPath)
+	if err != nil {
+		return fmt.Errorf("unable to backup file\n%w", err)
```
Comment (pivotal-david-osullivan) [inline sherpa/sbom.go:60]: unable to read XML file for conversion?

File: sherpa/sbom.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+	}
+
+	if err := b.writeJSONBOM(inputPath, bom); err != nil {
+		return fmt.Errorf("unable to backup file\n%w", err)
```
Comment (pivotal-david-osullivan) [inline sherpa/sbom.go:64]: unable to write converted JSON BOM file?

Comment (dmikusa-pivotal) [reply sherpa/sbom.go:64]:
```suggestion
		return fmt.Errorf("unable to write converted JSON BOM file\n%w", err)
```

File: sherpa/sbom.go (hunk 1 above)
Comment (dmikusa-pivotal) [reply sherpa/sbom.go:60]: Lol, I was clearly doing too much copy & paste here.

Comment (dmikusa-pivotal) [reply sherpa/sbom.go:60]:
```suggestion
		return fmt.Errorf("unable to read XML file for conversion\n%w", err)
```

**Pull request merged**